src/utils/genericUtils.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ *
+ * @param date
+ * @returns string
+ *
+ * This function is used to parse a date in any format into a string.
+ */
+export const parseDate = (date: unknown, includeTime = false): string => {
+  if (
+    typeof date === "string" ||
+    typeof date === "number" ||
+    date instanceof Date
+  ) {
+    return (
+      new Date(date).toLocaleDateString() +
+      (includeTime ? " " + new Date(date).toLocaleTimeString() : "")
+    );
+  } else {
+    console.error("Invalid date:", date);
+    return "N/A";
+  }
+};
+
+/**
+ *
+ * @param key
+ * @returns string
+ *
+ * This function is used to beautify a key string.
+ */
+export const beautifyKey = (key: string): string => {
```

Comment:
I would rename this function to something like `titleFromId` as I had to go and find some usage of the function before I really understood what it was for and it is the title that is passed in. The text description could be a bit better to indicate that your taking an ID/Title that has underscores/hyphens and converts that to a space and then uppercasing each word.